src/org/labkey/test/components/html/ToggleButton.java

```diff
@@ -0,0 +1,91 @@
+package org.labkey.test.components.html;
+
+import org.labkey.test.Locator;
+import org.labkey.test.WebDriverWrapper;
+import org.labkey.test.components.WebDriverComponent;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+public class ToggleButton extends WebDriverComponent<ToggleButton.ElementCache>
+{
+    final WebElement _el;
+    final WebDriver _driver;
+
+    protected ToggleButton(WebElement element, WebDriver driver)
+    {
+        _el = element;
+        _driver = driver;
+    }
+
+    @Override
+    public WebElement getComponentElement()
+    {
+        return _el;
+    }
+
+    @Override
+    public WebDriver getDriver()
+    {
+        return _driver;
+    }
+
+    public ToggleButton set(boolean enabled)
+    {
+        String desiredState = enabled ? "enabled" : "disabled";
+        if (get() != enabled)
+            getComponentElement().click();
+        WebDriverWrapper.waitFor(()-> get() == enabled,
+                "the toggle button did not become " + desiredState, 2000);
+        return this;
+    }
+
+    public boolean get()
+    {
+        return !getComponentElement().getAttribute("class").contains("off");
+    }
+
+
+    @Override
+    protected ElementCache newElementCache()
+    {
+        return new ElementCache();
+    }
+
+
+    protected class ElementCache extends WebDriverComponent.ElementCache
+    {
+
+    }
```

Comment:
ElementCache stuff is just taking up space. There's nothing to cache and there likely never will be.